### DISCUSSION TITLE
Does ort supports mixed array types for onnx session.run

### DISCUSSION THREAD

**Discussion by mush42** [pykeio/ort#26] [Q&A]
Hi @sudo-carson 

A project I'm involved with wants to adopt `ort`. But there is a concern about support for mixed array types as model inputs.

Is there any way to provide mixed array types to session.run?

Best
Musharraf

**Comment by decahedron1** [comment]
If by mixed array types you mean 2 inputs with different types, then yes, that is supported.

**Comment by mush42** [comment]
@synesthesiam would you mind chiming in?

**Comment by synesthesiam** [comment]
Thanks @sudo-carson! It looks like the input tensors to a session can have different dimensions as well. This was a big blocker for me with `onnxruntime-rs`.

I plan to port [larynx 2](https://github.com/rhasspy/larynx2/) to Rust using `ort`. This is a local text to speech system with growing language support (soon Arabic thanks to @mush42).

Out of curiosity, is `arm32v7` supported with `ort`?

**Reply by decahedron1** [reply]
> Out of curiosity, is `arm32v7` supported with `ort`?

Not at the moment, but if ONNX Runtime supports it, I'd be happy to add support in `ort` as well 🙂

**Reply by synesthesiam** [reply]
I doubt they ever will officially, but I've compiled my own version in the past. I plan to update it for 1.14, which you'd be welcome to use if you're interested.
